webapp/loggedin/modules/users/user/userViewState.js

```diff
@@ -91,10 +100,40 @@ export const useUserViewState = props => {
         group: !isUserAcceptPending && Authorizer.canEditUserGroup(user, surveyInfo, userToUpdate),
       })
 
+      ready.current = true
       enableValidation()
     }
   }, [loaded])
 
+
+  // PROFILE PICTURE
+
+  const [profilePicture, setProfilePicture] = useState(null)
+  const prevProfilePicture = usePrevious(profilePicture)
+  const pictureChanged = useRef(false)
+
+  useEffect(() => {
+    const fd = new FormData()
+    for (var key in formObject) {
+      fd.append(key, formObject[key])
+    }
+
+    if (prevProfilePicture && prevProfilePicture !== profilePicture) {
+      pictureChanged.current = true
+      // Only send the picture file if it has been edited by the user
+      fd.append('picture', profilePicture)
+    }
+
+    setFormData(fd)
+  }, [profilePicture, formObject])
+
+  const putRequest = useAsyncPutRequest(
```

Comment:
can we define a useAsyncPutMultipartRequest hook handling the right headers? And why is this request in the component and not in the action?

Reply:
Done. It's in the component because this component doesn't use redux